src/EnergyPlus/ScheduleManager.cc

```diff
@@ -1661,6 +1809,63 @@ namespace ScheduleManager {
 			hourlyFileValues.deallocate();
 		}
 
+		std::string curName;
+		Array1D< Real64 > hourlyColumnValues;
+		for ( auto& NameValue : CSVAllColumnNames ) {
+			curName = NameValue.first + "_shading";
+			hourlyColumnValues = CSVAllColumnNameAndValues[ NameValue.second ];
+			IsNotOK = false;
+			IsBlank = false;
+			VerifyName( curName, Schedule( {1,NumSchedules} ), SchNum, IsNotOK, IsBlank, CurrentModuleObject + " Name" );
+			if ( IsNotOK ) {
+				ErrorsFound = true;
+				if ( IsBlank ) curName = "xxxxx";
+			}
+			++SchNum;
+			Schedule( SchNum ).Name = curName;
+			Schedule( SchNum ).SchType = ScheduleInput_file;
+			
+			iDay = 0;
+			hDay = 0;
+			ifld = 0;
+			while ( true ) {
+				// create string of which day of year
+				++iDay;
+				++hDay;
+				if ( iDay > 366 ) break;
+				ExtraField = RoundSigDigits( iDay );
+				// increment both since a week schedule is being defined for each day so that a day is valid
+				// no matter what the day type that is used in a design day.
+				++AddWeekSch;
+				++AddDaySch;
+				// define week schedule
+				WeekSchedule( AddWeekSch ).Name = curName + "_shading_wk_" + ExtraField;
+				// for all day types point the week schedule to the newly defined day schedule
+				for ( kDayType = 1; kDayType <= MaxDayTypes; ++kDayType ) {
+					WeekSchedule( AddWeekSch ).DaySchedulePointer( kDayType ) = AddDaySch;
+				}
+				// day schedule
+				DaySchedule( AddDaySch ).Name = curName + "_shading_dy_" + ExtraField;
+				DaySchedule( AddDaySch ).ScheduleTypePtr = Schedule( SchNum ).ScheduleTypePtr;
+				// schedule is pointing to the week schedule
+				Schedule( SchNum ).WeekSchedulePointer( iDay ) = AddWeekSch;
+				// MinutesPerItem == 60
+
+				for ( jHour = 1; jHour <= 24; ++jHour ) {
+					++ifld;
+					curHrVal = hourlyColumnValues( ifld ); // hourlyFileValues((hDay - 1) * 24 + jHour)
+					for ( TS = 1; TS <= NumOfTimeStepInHour; ++TS ) {
+						DaySchedule( AddDaySch ).TSValue( TS, jHour ) = curHrVal;
+					}
+				}
+				if ( iDay == 59 && ScheduleFileShadingLeapYear ) { // 28 Feb
+					// Dup 28 Feb to 29 Feb (60)
+					++iDay;
+					Schedule( SchNum ).WeekSchedulePointer( iDay ) = Schedule( SchNum ).WeekSchedulePointer( iDay - 1 );
+				}
+			}
+		}
+
```

Comment:
Yeah I didn't see in there where you actually added JSON.  What I had requested previously was for the code to be set up so that once we added JSON, it would be *very easy* to modify this code to also read JSON inputs.  But as the JSON world isn't in here yet, we need to remove the option from the IDD and docs. (Unless I somehow missed the JSON reading in here and it actually does work...)

Reply:
No. Not ready. My misunderstanding. But yes here it is for when we can actually add JSON in the future.